File: apps/codemap-search/tests/fixtures/navigation/java/policy.java
import java.util.List;

class OrderPolicy {
    private final List<Rule> rules;

    OrderPolicy(List<Rule> rules) {
        this.rules = rules;
    }

    List<String> evaluate(Order order) {
        List<String> failures = rules.stream()
                .filter(rule -> !rule.check(order))
                .map(Rule::name)
                .toList();
        if (failures.isEmpty()) {
            return List.of();
        }
        return failures;
    }
}

File: apps/codemap-search/tests/fixtures/navigation/kotlin/policy.kt
import com.example.Rule

class OrderPolicy(private val rules: List<Rule>) {
    fun evaluate(order: Order): List<String> {
        val failures = rules.filter { rule -> !rule.check(order) }
        val names = failures.map { rule -> rule.name }
        if (failures.isEmpty()) {
            return emptyList()
        }
        return names
    }
}
